design-documents/graph-ql/framework/auth-annotation.md

```diff
@@ -0,0 +1,62 @@
+# Overview
+
+In the scope of adding features to the GraphQL schema, a client like PWA needs to know what queries need authentication.
+This is needed to know to refresh a token each time it expires, so this logic or what queries need a token doesn't get hardcoded into the client side.
+It is also useful to implement the "soft" auth mode for cart, when a token expires, and the client would still store some info from the last sesson client side.
+
+## Annotation
+
+We need add an anotatoin like `@auth` to support metadata that would be needed by the client:
+
+```graphql
+type Query {
+    customerCart: Cart! @auth(required:true)
```

Comment:
How do you envision the client-side using this directive? Is it primarily there as documentation, or is the client expected to do some magic with it?